### ISSUE TITLE
New focus appearance wording can prohibit fading focus indicators

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag#2333]
In a recent update to 2.4.11 Focus appearance, the "minimum area" exception was updated to say:

> has a contrast ratio of at least 3:1 against adjacent colors, or is no thinner than 2 CSS pixels.

This creates an odd scenario where a single pixel focus indicator can pass, but if that same focus indicator has a shadow around it the indicator fails. Here's a codepen showing the scenario: https://codepen.io/wilcofiers/full/wvpVNRZ

```html
<style>
  a:focus {
    outline: solid 1px #888;
    outline-offset: 0;
  }
  .fail:focus { box-shadow: 0 0 2px 1px #888; }
</style>
<a href="#">Passed</a>
<a href="#" class="fail">Failed</a>
```

The passed example here is fine because it meets the default scenario. It has sufficient contrast, and it encloses the component.

For the failed case, because of the box shadow, a part of the focus indicator doesn't meet the 3:1 contrast ratio, so the default case isn't met. Looking further to the exceptions, the area of the focus indicator that has sufficient contrast is the outline. So we'll go through the requirements for the exception. Bullet 1, the area is sufficient. Bullet 2, the contrast with the unfocused state is sufficient.

But then bullet 3, firstly the outline is only 1px thick, we can not count the box-shadow as the second pixel thickness since the box-shadow doesn't have enough contrast between the focused and unfocused state. So then the outline needs to have enough contrast with all its surrounding colors. There are two colors touching the outline, the box-shadow on the outside, and the white background on the inside. And so, because the contrast between the outline and box-shadow is insufficient this exception isn't met.

That's different from the previous wording where the contrast requirement only applied to colors of the component.

**Comment by mbgower** [comment]
Thanks for surfacing!
First off, at the moment, the example has the indicator abutting the link underline, so the Passed link is going to fail the 3rd bullet of the primary, right? 
![image](https://user-images.githubusercontent.com/14298245/165811675-8ef50659-2194-496b-84d6-3008e75e5859.png)

We need to offset the 1px indicator so it does not abut any of the link.
![image](https://user-images.githubusercontent.com/14298245/165812073-21c60a30-2a01-4a98-8048-34119b9947f6.png)

It doesn't really change the situation, but I just wanted to make sure we had a passing example. Assuming we're agreed on that, I see a few ways to potentially tackle, off the top of my head:
1. change the wording to "has a contrast ratio of at least 3:1 against adjacent non-focus-indicator colors, or is no thinner than 2 CSS pixels." Or "has a contrast ratio of at least 3:1 against the indicator's background color, or is no thinner than 2 CSS pixels"
2. returning to what it said before: "has a contrast ratio of at least 3:1 against the component, or is no thinner than 2 CSS pixels"
3. slightly reword the prior wording to something like: "has a contrast ratio of at least 3:1 against a directly adjacent component, or is no thinner than 2 CSS pixels.
4. leave as is, and accept some edge case failures that are still easy to resolve

The first suggestion would mean for adjacent contrast, we're always just assessing the focus indicator against stuff abutted that is NOT part of the focus indicator. We've already established that part of the indicator 1) must be big enough, and 2) must have pixels that contrast sufficiently between focused and not focused, so I _think_ that can address this without having a negative consequence? Does that address? Do you prefer either wording? The second one draws on wording already used in a user agent exception.

For option 2, my problem with the prior wording is similar to your concerns in the current situation. You can arrive at something that should passes -- a 1px offset outline -- but if the indicator is the same color as the component, it fails the third sub-bullet! I really don't think we want to return to that.

Option 3 is an attempt to resolve option 2 in a very contained way -- with focus on the focus indicator versus only the component. This goes back to the original intent -- if focus indicator abuts the component with insufficient contrast, make it thick enough to be more obvious. My slight concern with this is that if the indicator was against something other than the indicator, it _might_ become tougher to see. Think of 3 closely positioned buttons where the indicator abuts the neighbours when a button gets focus. You would effectively get an odd situation when only the top and bottom indicators would be very discernible. This may be uncommon enough that we don't care.

Option 4 is not something I'm advocating here, but I did want to note that it is easy to resolve the example with a 2px indicator 
![image](https://user-images.githubusercontent.com/14298245/165814504-5e17238d-e601-42ef-b3b3-e6f8a635de43.png)

**Comment by alastc** [comment]
Just noting that this is already [in the survey](https://www.w3.org/2002/09/wbs/35422/wcag22-focus-appearance-enhanced2/#wbsq49).

**Comment by mbgower** [comment]
I created a [PR that addresses](https://github.com/w3c/wcag/pull/2341), based on discussion from 2.x backlog meeting.

**Comment by alastc** [comment]
Hi @WilcoFiers, #2341 has been merged so I'm closing this.
